### PR TITLE
Add option grpc.RequestValidator

### PR DESCRIPTION
**Pull request by peter-edge**

### CODE REVIEW

File: transport/grpc/options.go [hunk 1]
```diff
@@ -83,6 +84,17 @@ func Logger(logger *zap.Logger) TransportOption {
 	}
 }
 
+// RequestValidator specifies an option to validate a transport.Request
+// before allowing an outbound call to be made or an inbound call
+// to be processed.
+//
+// By default, no validation is done.
+func RequestValidator(requestValidator func(*transport.Request) error) TransportOption {
```
Comment (AlexAPB) [inline transport/grpc/options.go]: Just a thought: If you can see the case for multiple request validators, we could maybe make them composable.

Comment (peter-edge) [reply transport/grpc/options.go]: Made it possible to use this option multiple times.

**Comment by akshayjshah** [comment]
For context, we're planning to use this functionality internally to throw errors when calling proto-encoded procedures over the gRPC transport. I have some reservations about this approach, but they're part of a discussion that goes a bit beyond this single PR. Can we hold off on merging this until the discussion on the internal "Towards gRPC and Protobuf" concludes?

**Comment by peter-edge** [comment]
I'm going to close this for now, just to clear it out on principle, we can reopen this when we finish up that discussion :-)

**Pull request closed without merging**